### PR TITLE
Make recalculating journals more robust

### PR DESCRIPTION
**Pull request by h-lame**
To follow up on comments on #416 which has already been merged.

### CODE REVIEW

**Comment by pixeltrix** [comment]
Will this make the queries significantly slower?

**Comment by h-lame** [comment]
Probably, but it'll also make them "correct" so I don't know if we have a choice :(.  There's still outstanding discussion about how to manage running these while the site is in use (https://github.com/alphagov/e-petitions/pull/416#issuecomment-164082304 and https://github.com/alphagov/e-petitions/pull/416#issuecomment-164391945).

File: app/models/country_petition_journal.rb [hunk 1]
```diff
@@ -40,6 +42,7 @@ def self.reset!
         timezone('utc', now()), timezone('utc', now())
       FROM signatures
       WHERE state = 'validated'
+      AND country <> ''
```
Comment (pixeltrix) [inline app/models/country_petition_journal.rb]: Won't this include signatures where country is NULL? Also don't we validate for presence of country so it shouldn't be empty anyway.

Comment (h-lame) [inline app/models/country_petition_journal.rb]: I guess I was playing safe, our schema allows for null `country` and I'm sure I read that in PSQL `<>` will act as both "NOT NULL" and "disallow blanks".  Could drop the condition though if we felt it was too much.

Comment (pixeltrix) [inline app/models/country_petition_journal.rb]: Hmm, appears to be true when I just tested it - can you rebase and then I'll merge it, thanks.

**Comment by h-lame** [comment]
rebased.

**Pull request merged**